terraform/dotnet/k8s/deploy/main.tf

```
@@ -0,0 +1,168 @@
+          --docker-server=https://${var.release_testing_ecr_account}.dkr.ecr.us-east-1.amazonaws.com \
+          --docker-username=AWS \
+          --docker-password="$${TOKEN}"
+
+        yq eval '.spec.template.spec.imagePullSecrets += [{"name": "release-testing-ecr-secret"}]' -i dotnet-frontend-service-depl.yaml
+        yq eval '.spec.template.spec.imagePullSecrets += [{"name": "release-testing-ecr-secret"}]' -i dotnet-remote-service-depl.yaml
+      fi
+
+      echo "LOG: Applying sample app deployment files"
+      kubectl apply -f dotnet-frontend-service-depl.yaml
+      kubectl apply -f dotnet-remote-service-depl.yaml
+
+      echo "Wait for sample app to be reach ready state"
+      sleep 10
+      kubectl wait --for=condition=Ready --request-timeout '10m' pod --all -n dotnet-sample-app-namespace
+
+      # Emit main and remote service pod IP
+      echo "LOG: Outputting remote service pod IP to SSM using put-parameter API"
+      aws ssm put-parameter --region ${var.aws_region} --name dotnet-main-service-ip-${var.test_id} --type String --overwrite --value $(kubectl get pod --selector=app=dotnet-sample-app -n dotnet-sample-app-namespace -o jsonpath='{.items[0].status.podIP}')
+      aws ssm put-parameter --region ${var.aws_region} --name dotnet-remote-service-ip-${var.test_id} --type String --overwrite --value $(kubectl get pod --selector=app=dotnet-remote-app -n dotnet-sample-app-namespace -o jsonpath='{.items[0].status.podIP}')
+
+      # Wait a bit more in case the sample apps aren't ready yet
+      sleep 30
+
+      # Deploy the traffic generator
+      kubectl create deployment -n dotnet-sample-app-namespace traffic-generator \
+        --image=$ACCOUNT.dkr.ecr.${var.aws_region}.amazonaws.com/e2e-test-resource:traffic-generator \
+        --replicas=1
+
+      # Patch it with ImagePull always policy so that it pulls the latest image from the ECR
+      kubectl patch deployment -n dotnet-sample-app-namespace traffic-generator --patch '{"spec": {"template": {"spec": {"containers": [{"name": "e2e-test-resource", "imagePullPolicy": "Always"}]}}}}'
+      kubectl patch deployment traffic-generator -n dotnet-sample-app-namespace --type='json' -p='[{"op": "add", "path": "/spec/template/spec/imagePullSecrets", "value": [{"name": "ecr-secret"}]}]'
+
+      # Add the appropriate environment variables to the traffic generator
+      kubectl set env -n dotnet-sample-app-namespace deployment/traffic-generator MAIN_ENDPOINT=$(kubectl get pods -n dotnet-sample-app-namespace --selector=app=dotnet-sample-app -o jsonpath='{.items[0].status.podIP}'):8080
+      kubectl set env -n dotnet-sample-app-namespace deployment/traffic-generator REMOTE_ENDPOINT=$(kubectl get pod -n dotnet-sample-app-namespace --selector=app=dotnet-remote-app -o jsonpath='{.items[0].status.podIP}')
+      kubectl set env -n dotnet-sample-app-namespace deployment/traffic-generator ID=${var.test_id}
```

Made this improvement based on Ping's suggestion [here](https://github.com/aws-observability/aws-application-signals-test-framework/pull/223#discussion_r1752405876). You should do the same
```suggestion
      kubectl set env -n dotnet-sample-app-namespace deployment/traffic-generator \
      MAIN_ENDPOINT=$(kubectl get pods -n dotnet-sample-app-namespace --selector=app=dotnet-sample-app -o jsonpath='{.items[0].status.podIP}'):8080 \
      REMOTE_ENDPOINT=$(kubectl get pod -n dotnet-sample-app-namespace --selector=app=dotnet-remote-app -o jsonpath='{.items[0].status.podIP}') \
      ID=${var.test_id}
```